### PR TITLE
[3.x] Remove the separator from `ItemList`'s thumbnails mode

### PR DESCRIPTION
**Pull request by Chubercik**
This PR brings parity with #82236 to the `3.x` branch.

### CODE REVIEW

File: scene/gui/item_list.cpp [hunk 1]
```diff
@@ -1155,13 +1155,16 @@ void ItemList::_notification(int p_what) {
...
+		// If not in thumbnails mode, draw visible separators.
+		if (icon_mode != ICON_MODE_TOP) {
+			for (int i = first_visible_separator; i < separators.size(); i++) {
+				if (separators[i] > clip.position.y + clip.size.y) {
+					break; // done
+				}
 
-			const int y = base_ofs.y + separators[i];
-			draw_line(Vector2(bg->get_margin(MARGIN_LEFT), y), Vector2(width, y), guide_color);
+				const int y = base_ofs.y + separators[I];
```
Comment (AThousandShips) [inline scene/gui/item_list.cpp:1165]:
```suggestion
				const int y = base_ofs.y + separators[i];
```

Comment (Chubercik) [reply scene/gui/item_list.cpp]: I'm sorry, I have no idea how that capital `i` got there..

**Comment by Chubercik** [comment]
Pre-patch and with the fix:

image image

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**